pkgs/development/libraries/glibc/nix-nss-open-files.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/nss/nss_files/files-XXX.c b/nss/nss_files/files-XXX.c
+index 1db9e46127..6cef6b2f4e 100644
+--- a/nss/nss_files/files-XXX.c
++++ b/nss/nss_files/files-XXX.c
+@@ -40,6 +40,7 @@
+ #define ENTNAME_r	CONCAT(ENTNAME,_r)
+ 
+ #define DATAFILE	"/etc/" DATABASE
++#define NIX_DATAFILE "NIX_etc_" DATABASE
```

Comment:
I am not so sure if it is a good idea to do this for all entries yet. Maybe we should start with `/etc/protocols` and `/etc/services` and than expand on demand. So I would propose to define `NIX_DATAFILE` in those files in uppercase and than do an `ifdef NIX_DATAFILE` in `internal_setent`.

Reply:
Thank you for the suggestion.